src/Aggregate/ConfigurableAggregateTranslator.php

```diff
@@ -211,4 +223,37 @@ public function extractPendingStreamEvents($eventSourcedAggregateRoot)
 
         return $recordedEvents;
     }
+
+    /**
+     * @param object $eventSourcedAggregateRoot
+     * @param Message[] $events
+     * @throws Exception\AggregateTranslationFailedException
+     */
+    public function applyPendingStreamEvents($eventSourcedAggregateRoot, array $events)
+    {
+        if (! is_object($eventSourcedAggregateRoot)) {
+            throw new AggregateTranslationFailedException('Event sourced Aggregate Root needs to be an object. Got ' . gettype($eventSourcedAggregateRoot));
+        }
+
+        if (! method_exists($eventSourcedAggregateRoot, $this->applyRecordedEventsMethodName)) {
+            throw new AggregateTranslationFailedException(
+                sprintf(
+                    'Can not apply recorded events to aggregate root %s. The AR is missing a method with name %s!',
+                    get_class($eventSourcedAggregateRoot),
+                    $this->applyRecordedEventsMethodName
+                )
+            );
+        }
+
+        foreach ($events as $event) {
+            if (!$event instanceof Message) {
+                throw new AggregateTranslationFailedException(sprintf(
+                    'A recorded event of the aggregate root %s has the wrong type. Expected Prooph\Common\Messaging\Message. Got %s',
+                    get_class($eventSourcedAggregateRoot),
+                    is_object($event)? get_class($event) : gettype($event)
+                ));
+            }
+            $eventSourcedAggregateRoot->{$this->applyRecordedEventsMethodName}($event);
```

Comment:
Before applying the event we need to check if a `messageToEventCallback` was injected in the translator. If so the callback needs to be invoked first.